cli/lib/kontena/cli/nodes/labels/remove_command.rb

```diff
@@ -5,14 +5,28 @@ class RemoveCommand < Kontena::Command
     parameter "NODE", "Node name"
     parameter "LABEL ...", "Labels"
 
+    option '--force', :flag, "Do not abort if items in label list are not found"
+
     requires_current_master
     requires_current_master_token
     requires_current_grid
 
     def execute
-      node = client.get("nodes/#{current_grid}/#{self.node}")
-      data = { labels: Array(node['labels']).reject {|label| label_list.include?(label) } }
-      client.put("nodes/#{node['id']}", data)
+      node_data = client.get("nodes/#{current_grid}/#{self.node}")
+
+      node_data['labels'] ||= []
+
+      found_labels = label_list.uniq & node_data['labels']
+
+      if !force? && found_labels.size != label_list.uniq.size
+        missing = label_list - found_labels
+        exit_with_error "Label#{'s' if missing.size > 1} #{pastel.cyan(missing.join(', '))} not found on node #{pastel.cyan(node)}"
+      end
+
+      return nil if found_labels.empty?
+
+      data = { labels: node_data['labels'] - found_labels }
```

Comment:
This could probably be simplified to just `{ labels: node_data['labels'] - label_list }`, but that /should/ be equivalent to `found_labels`.